libqaul/common/src/message.rs

```diff
@@ -0,0 +1,16 @@
+//! Types for representing a message to be sent between users.
+use crate::UserID;
+use crate::payload::Payload;
+
+/// A `Message` to be sent from the `sender` to the `recipient`, associated
+/// with the given `service`.
+/// Differs from a RATMAN message; this is a user-to-user, fully encapsulated
+/// message which does not care (or know) how it is delivered, merely that it
+/// was sent by one user and is meant for another.
+pub struct Message {
+    sender: UserID,
+    recipient: UserID,
+    // TODO: Add some kind of cryptographic signing here
```

Comment:
I suggest we just add a new type for now, called `MsgSignature` (name up for :bike::house:), that contains a `u32` for a unix timestamp. We can later amend it with whatever format our signatures will have.

It's a bit of an open question. The old codebases used `mbedtls` but I haven't looked at the Rust crate for it yet and don't know how nice it is to work with. There's also a bunch of alternatives to work with. Plus, I kinda got it into my head that maybe we could implement the Axolotl ratchet.

**TLDR:** Maybe add that `MsgSignature` type into that file, add it as a field, just to make sure we don't forget it. The rest can be seen later and we an merge this :)

Reply:
Will do!